src/lfortran/semantics/ast_body_visitor.cpp

```diff
@@ -455,6 +456,166 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
         }
     }
 
+    void visit_DataImpliedDo(const AST::DataImpliedDo_t &x) {
+        std::string loop_var_name = to_lower(x.m_var);
+        // auto sym = current_scope->resolve_symbol(array);
+        // if (sym != nullptr) throw SemanticError("Data statement loop variable cannot be have same name as other variable.", x.base.base.loc);
+        auto start = AST::down_cast<AST::Num_t>(x.m_start);
```

Comment:
IMHO, we should use full type name instead of `auto`. I prefer verbosity over ease of typing.

Reply:
I don't really care about ease of typing, it's much more easy to read I think. But sure, I'll change it once I actually open a PR.